### PR TITLE
fix: control chars on progress listener for Windows OS

### LINKED ISSUE #352: [usability] Spinner for build and deploy commands does not look correct on Windows
Sample output:

```
Microsoft Windows [Version 10.0.19042.906]
(c) Microsoft Corporation. All rights reserved.

C:\Users\User>func build -p functions/gofunc

←[1A←[K   Building function image
←[1A🕒 ←[1B

```

### PR DESCRIPTION
**Pull request by jrangelramos**
# Changes

- :broom: Cleaning spinner for build and deploy commands for Windows platform

/kind cleanup

Fixes #352

### CODE REVIEW

File: progress/progress.go [hunk 1]
```diff
@@ -237,7 +241,11 @@ func (b *Bar) spin(ch <-chan time.Time) {
 		// Writes the spinner frame at the beginning of the previous line, moving
 		// the cursor back to the beginning of the current line for any errors or
 		// informative messages.
-		fmt.Fprintf(b.out, "\r%v%v%v\r", up, spinner[idx], down)
+		if runtime.GOOS == "windows" {
+			fmt.Fprintf(b.out, "\r%v\r", spinner[idx])
```
Comment (lance) [inline progress/progress.go:245]: How does this end up looking on Windows? I don't even have a VM at the moment to try it out.

Comment (jrangelramos) [reply progress/progress.go:245]: Hi @lance, it looks like this:

https://user-images.githubusercontent.com/26138643/131407735-1c1476c1-2e04-490d-83b6-4196bea36975.mp4

Comment (jrangelramos) [reply progress/progress.go:245]: Note spinner is rendered on next line of the message. Previous code uses control chars to move it up (before rendering the spinner) and then move it down, but they don't work on Windows and output gets like:
```
←[1A←[K   Building function image
←[1A/←[K
```

**Pull request merged**